flutter/news/lib/features/frontpage/presentation/bloc/top_headlines_state.dart

```diff
@@ -0,0 +1,29 @@
+part of 'articles_cubit.dart';
+
+abstract class ArticlesState extends Equatable {
```

Comment:
https://pub.dev/packages/freezed

@Diogo00dev could we use this? can we explore it? It has support for `Union/Sealed Classes` for dart, and that could be something great great to have

Reply:
I also thing It's a good idea. 
One thing to keep in mind is that you would need to generate these files when you modify them (or keep the autogeneration into "watch" mode). 

On the positive side you don't need to implement equatable manually, you can also use it for JSON encoding/decoding and for Union/Sealed Classes

Reply:
Let's try with freeze @Diogo00dev , I might give it a try for my next PR if you don't mind it